### PR TITLE
add AIOSEO pack placeholder tags processing (WP-736)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Helpers/ShortcodeHelper.php [hunk 1]
```diff
@@ -264,7 +260,6 @@ private function replaceHandlerForMining(array $shortcodeList)
     protected function preUploadFiltering($name, $attributes)
     {
         $preparedAttributes = static::maskAttributes($name, $attributes);
-        $this->postReceiveFiltering($preparedAttributes);
```
Comment (vsolovei-smartling) [inline inc/Smartling/Helpers/ShortcodeHelper.php:267]: Function result unused

**Review by PavelLoparev — COMMENTED** [review]
I probably need some visual demo for this. It's hard to say what was wrong and what fix is.

Also, it would be nice to have a set of tests for:
1. `splitTagField`
2. `convertTagFieldParts`

to test all the corner cases (there are a lot of conditions inside)

File: inc/Smartling/Helpers/ShortcodeHelper.php (hunk 1 above)
Comment (PavelLoparev) [reply inc/Smartling/Helpers/ShortcodeHelper.php:267]: There could be surprise - modified passed `$preparedAttributes` by reference)

File: inc/Smartling/Helpers/XmlHelper.php [hunk 2]
```diff
@@ -79,11 +79,12 @@ private static function setTranslationComments(DOMDocument $document)
             'Smartling Wordpress Connector version: ' . Bootstrap::getCurrentVersion(),
             'Wordpress installation host: ' . Bootstrap::getHttpHostName(),
             vsprintf(
-                ' smartling.placeholder_format_custom = %s ',
+                'smartling.placeholder_format_custom = %s',
                 [
                     ShortcodeHelper::getMaskRegexp(),
                 ]
             ),
+            'smartling.placeholder_format_custom = #sl-start#.+#sl-end#',
```
Comment (dimitrystd) [inline inc/Smartling/Helpers/XmlHelper.php:87]:
```suggestion
            'smartling.placeholder_format_custom = #sl-start#.+?#sl-end#',
```

**Comment by vsolovei-smartling** [comment]
> I probably need some visual demo for this. It's hard to say what was wrong and what fix is.
> 
> Also, it would be nice to have a set of tests for:
> 
> 1. `splitTagField`
> 2. `convertTagFieldParts`
> 
> to test all the corner cases (there are a lot of conditions inside)

I've added a test, and while the total class coverage is not 100%, both the splitTagField and joinTagField are covered

**Review by PavelLoparev — APPROVED** [review]

File: inc/Smartling/Helpers/XmlHelper.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Helpers/XmlHelper.php:87]: This is what I've been looking for in all the wrong places, thanks

File: inc/Smartling/ContentTypes/ExternalContentAioseo.php [hunk 3]
```diff
@@ -113,6 +121,11 @@ public function setContentFields(array $content, SubmissionEntity $submission):
                 $content[$field] = json_encode($content[$field], JSON_THROW_ON_ERROR);
             }
         }
+        foreach ($this->tagFields as $field) {
+            if (array_key_exists($field, $content)) {
+                $content[$field] = $this->removePlaceholders($content[$field]);
```
[the remark this answers is not in the document]
Comment (vsolovei-smartling) [reply inc/Smartling/ContentTypes/ExternalContentAioseo.php:126]: Done

File: inc/Smartling/Helpers/ShortcodeHelper.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Helpers/ShortcodeHelper.php:267]: No action required, function does not alter arguments in any way, safe to remove
